WoW-Realm-Tracker/ViewControllers/FavoritesViewController.swift

```diff
@@ -83,8 +84,8 @@ class FavoritesViewController: UITableViewController {
 }
 
 extension FavoritesViewController: RealmsDelegate {
-    func receivedRealms(realms: [Realm]) {
-        controller = FavoriteRealmsController(realms: realms)
+    func receivedRealms(response: Result<[Realm], NSError>) {
+        response.map { realms in self.controller = FavoriteRealmsController(realms: realms) }
```

Comment:
I just tried giving something like that a shot but I'm getting `Enum case "Some" not found in type Result<FavoriteRealmsController, NSError>`

I understand and agree with what your saying about avoiding map for side effects and ignoring the return value. Would the best use of `Result<[Realm], NSError>` here be to do something like:

```swift
switch response {
  case .Success(let realms): loadRealms(realms)
  case .Failure(let error): print(error)
}
```